contracts/coordinator/src/state.rs

```diff
@@ -220,12 +217,22 @@ pub fn save_deployed_contracts(
 
 pub fn deployed_contracts(
     storage: &dyn Storage,
-    deployment_name: nonempty::String,
-) -> Result<ChainContracts, Error> {
-    DEPLOYED_CHAINS
-        .may_load(storage, deployment_name.to_string())
-        .change_context(Error::StateParseFailed)?
-        .ok_or(report!(Error::DeploymentNameNotFound(deployment_name)))
+    start: Option<nonempty::String>,
+    limit: Option<u64>,
+) -> Result<Vec<ChainContracts>, Error> {
+    let start = start.map(Bound::inclusive);
+
+    Ok(match limit {
+        Some(l) => DEPLOYED_CHAINS
+            .range(storage, start, None, Order::Ascending)
+            .take(usize::try_from(l).change_context(Error::StateParseFailed)?)
+            .filter_map(|entry| entry.ok().map(|(_, contracts)| contracts))
+            .collect(),
```

Comment:
you can get around the two cases by setting limit to u64::Max if `None`. What could be interesting though is to treat the case `Some(1)` separately and use `may_load` instead of `range` to save on overhead